client/web/src/user/settings/auth/ExternalAccountSignIn.test.tsx

```diff
@@ -0,0 +1,87 @@
+import {Attribute, getUsernameOrEmail, SamlExternalData} from './ExternalAccountsSignIn'
+
+function toAttribute(value: string): Attribute {
+    return {
+        Name: 'test',
+        Values: [
+            {
+                Value: value,
+            }
+        ]
+    }
+}
+
+function samlDataObject(keysValues: object): SamlExternalData {
+    // Add some other fields to make sure we are getting the right ones
+    Object.assign(keysValues, {'any field':'false'})
+    Object.assign(keysValues, {'random field':'Mon Oct 10 2022 13:07:34 GMT+0000 (Coordinated Universal Time)'})
+    Object.assign(keysValues, {'one more random field':'banana'})
+
+    const testData: unknown = {
+        'Values': Object.fromEntries(
+            Object.entries(keysValues).map(([key, value]) =>
+                [key, toAttribute(value)]
+            )
+        )
+    }
+
+    return testData as SamlExternalData
+}
```

Comment:
Can be IMO simplified to:
```suggestion
function samlDataObject(testData: object): SamlExternalData {
    // Add some other fields to make sure we are getting the right ones
    testData['any field'] = 'false'
    testData['random field'] = 'Mon Oct 10 2022 13:07:34 GMT+0000 (Coordinated Universal Time)'
    testData['one more random field'] = 'banana'

    return {
      Values: testData
    }
}
```

Reply:
Tried, but got a `Element implicitly has an 'any' type because expression of type 'string' can't be used to index type` error. So will keep it how it is.